From 67c8ace3c963a56621de3a0ec5dafef31bedc7a7 Mon Sep 17 00:00:00 2001
From: Peter Zijlstra <peterz@infradead.org>
Date: Thu, 3 Aug 2023 10:43:22 +0200
Subject: [PATCH 10/23] smp: Mark nr_cpu_ids __ro_after_init

The variable nr_cpu_ids contains the number of bits set in
cpu_possible_mask and is thus constant after init.

Mark the variable __ro_after_init to both improve performance and
enhance security, since the variable is exported. A module writing to
it would be 'unfortunate'.

Signed-off-by: Peter Zijlstra (Intel) <peterz@infradead.org>
---
 kernel/smp.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/kernel/smp.c b/kernel/smp.c
index 385179dae..5726c79b5 100644
--- a/kernel/smp.c
+++ b/kernel/smp.c
@@ -937,7 +937,7 @@ early_param("maxcpus", maxcpus);
 
 #if (NR_CPUS > 1) && !defined(CONFIG_FORCE_NR_CPUS)
 /* Setup number of possible processor ids */
-unsigned int nr_cpu_ids __read_mostly = NR_CPUS;
+unsigned int nr_cpu_ids __ro_after_init = NR_CPUS;
 EXPORT_SYMBOL(nr_cpu_ids);
 #endif
 
-- 
2.42.0

